src/network.c

```diff
@@ -791,20 +816,32 @@ void close_connection(Connection *conn) {
  */
 int setup_SSL_connection(Connection *conn, SSL_CTX *ctx) {
   int ssl_err;
+  int ssl_ret;
+  int ssl_errno;
+  ERR_clear_error();
   conn->ssl = SSL_new(ctx);
   if (conn->ssl == NULL) {
     log_println(4, "SSL_new failed");
     return ENOMEM;
   }
+  ERR_clear_error();
   if (SSL_set_fd(conn->ssl, conn->socket) == 0) {
     log_println(4, "SSL_set_fd failed");
     return EIO;
   }
-  if ((ssl_err = SSL_accept(conn->ssl)) != 1) {
-    ssl_err = SSL_get_error(conn->ssl, ssl_err);
-    log_println(4, "SSL_accept failed: %s (%d)", ssl_error_str(ssl_err),
-                ssl_err);
-    return EIO;
-  }
+  do {
+    ssl_err = 0;
+    ERR_clear_error();
+    ssl_ret = SSL_accept(conn->ssl);
+    if (ssl_ret != 1) {
+      ssl_err = SSL_get_error(conn->ssl, ssl_ret);
+      ssl_errno = errno;
```

Comment:
For consistency with the error handling around `SSL_read` and `SSL_write`, please place `ssl_errno = errno` immediately following `SSL_accept`.